### PR TITLE
km plot ggplot

### PR DESCRIPTION
**Pull request by MikeJSeo**

### CODE REVIEW

File: R/plot_km2.R [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+#' @param xlim x limit for the x-axis of the plot
+#' @param use_colors a character vector of length up to 4, colors to the KM curves, it will be passed to 'col' of \code{lines()}
+#' @param use_line_types a numeric vector of length up to 4, line type to the KM curves, it will be passed to 'lty' of \code{lines()}
+
+basic_kmplot2 <- function(kmlist,
+                          kmlist_name,
+                          endpoint_name = "Time to Event Endpoint",
+                          show_risk_set = TRUE,
+                          main_title = "Kaplan-Meier Curves",
+                          break.x.by = NULL,
```
Comment (gravesti) [inline R/plot_km2.R:229]:
```suggestion
                          break_x_by = NULL,
```
for consistency?

File: R/plot_km2.R [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+  }
+
+  # Produce the Kaplan-Meier plot
+  survminer_plot <- survminer::ggsurvplot(kmlist,
+    linetype = use_line_types,
+    palette = use_colors,
+    size = 0.2,
+    combine = TRUE,
+    risk.table = show_risk_set,
+    risk.table.y.text.col = T,
```
Comment (gravesti) [inline R/plot_km2.R:249]:
```suggestion
    risk.table.y.text.col = TRUE,
```

File: R/plot_km2.R [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+    risk.table.y.text = FALSE,
+    break.x.by = break.x.by,
```
Comment (gravesti) [inline R/plot_km2.R:251]:
```suggestion
    break.x.by = break_x_by,
```

File: R/plot_km2.R [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+    censor = censor,
+    censor.size = 2,
+    xlab = "Time",
+    ylab = endpoint_name,
+    legend.title = "Treatment",
+    legend = c(0.85, 0.82),
+    title = paste0(main_title, "\nEndpoint:", endpoint_name),
+    legend.labs = kmlist_name,
+    tables.theme = theme_cleantable(),
+    ggtheme = theme_classic(base_size = 10),
```
Comment (gravesti) [inline R/plot_km2.R:261]:
```suggestion
    ggtheme = ggplot2::theme_classic(base_size = 10),
```

**Review by gravesti — COMMENTED** [review]
@MikeJSeo Looks nice! I've noted a few small things to fix and generally I wonder if the code can be more compact with lapply or something. Let me know if I can help :)

File: R/plot_km2.R [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+  trt_var_agd <- toupper(trt_var_agd)
+
+  # pre check
+  if (!"maicplus_estimate_weights" %in% class(weights_object)) stop("weights_object should be an object returned by estimate_weights")
+  if (!all(c("USUBJID", "TIME", "EVENT", trt_var_ipd) %in% names(tte_ipd))) stop(paste("tte_ipd needs to include at least USUBJID, TIME, EVENT,", trt_var_ipd))
+  if (!all(c("TIME", "EVENT", trt_var_agd) %in% names(tte_pseudo_ipd))) stop(paste("tte_pseudo_ipd needs to include at least TIME, EVENT,", trt_var_agd))
+  km_layout <- match.arg(km_layout, choices = c("all", "by_trial", "by_arm"), several.ok = FALSE)
+
+  # preparing data
+  is_anchored <- ifelse(is.null(trt_common), FALSE, TRUE)
```
[the remark this answers is not in the document]
Comment (MikeJSeo) [reply R/plot_km2.R:55]: This is from Greg's code which I copied from plot_km. I have made changes in plot_km as well.

File: R/plot_km2.R [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+  tte_ipd <- tte_ipd[tte_ipd[[trt_var_ipd]] %in% c(trt_ipd, trt_common), , drop = TRUE]
```
[the remark this answers is not in the document]
Comment (MikeJSeo) [reply R/plot_km2.R:56]: @hoppanda Greg can you comment on this?

File: R/plot_km2.R [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+      splots[[2]] <- basic_kmplot2(kmlist3,
+        kmlist3_name,
+        main_title = paste0("Kaplan-Meier Curves \n(", trt_ipd, " vs ", trt_agd, ")"),
+        ...
+      )
+      splots[[4]] <- basic_kmplot2(kmlist4,
+        kmlist4_name,
+        main_title = paste0("Kaplan-Meier Curves of Common Comparator \n", trt_common, "(IPD vs AgD Trial)"),
+        ...
+      )
```
[the remark this answers is not in the document]
Comment (MikeJSeo) [reply R/plot_km2.R:202]: Thanks for the suggestion. I have made a change using lapply.

**Comment by gravesti** [comment]
@MikeJSeo Can you see if you can fix any of the checks? I think we need to add some lines like this to the top of the new functions
```
if (!requireNamespace("survminer", quietly = TRUE)) stop("survminer package is required for this function")
```

File: R/plot_km2.R (hunk 6 above)
[the remark this answers is not in the document]
Comment (gravesti) [reply R/plot_km2.R:56]: Confirmed change to `drop = FALSE`

File: R/plot_km2.R [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+  # pre check
+  if (!"maicplus_estimate_weights" %in% class(weights_object)) stop("weights_object should be an object returned by estimate_weights")
+  if (!all(c("USUBJID", "TIME", "EVENT", trt_var_ipd) %in% names(tte_ipd))) stop(paste("tte_ipd needs to include at least USUBJID, TIME, EVENT,", trt_var_ipd))
+  if (!all(c("TIME", "EVENT", trt_var_agd) %in% names(tte_pseudo_ipd))) stop(paste("tte_pseudo_ipd needs to include at least TIME, EVENT,", trt_var_agd))
+  km_layout <- match.arg(km_layout, choices = c("all", "by_trial", "by_arm"), several.ok = FALSE)
+
+  # preparing data
+  is_anchored <- ifelse(is.null(trt_common), FALSE, TRUE)
+  tte_ipd <- tte_ipd[tte_ipd[[trt_var_ipd]] %in% c(trt_ipd, trt_common), , drop = TRUE]
+  tte_pseudo_ipd <- tte_pseudo_ipd[tte_pseudo_ipd[[trt_var_agd]] %in% c(trt_agd, trt_common), , drop = TRUE]
```
[the remark this answers is not in the document]
Comment (gravesti) [reply R/plot_km2.R:57]: Change here to `drop = FALSE`

**Comment by MikeJSeo** [comment]
@gravesti I think all checks have passed. I won't be pushing any more changes if you want to play around with it.

**Comment by gravesti** [comment]
Thanks for the hard work on that @MikeJSeo !

**Review by gravesti — APPROVED** [review]

**Pull request merged**